bundle/config/mutator/select_target.go

```diff
@@ -49,9 +49,5 @@ func (m *selectTarget) Apply(_ context.Context, b *bundle.Bundle) diag.Diagnosti
 	// TODO: remove when Environments section is not supported anymore.
 	b.Config.Bundle.Environment = b.Config.Bundle.Target
 
-	// Clear targets after loading.
-	b.Config.Targets = nil
-	b.Config.Environments = nil
-
```

Comment:
Instead of keeping these around here, could you break out a field on the bundle struct where we can keep a snapshot of the selected target? Then you can interrogate it and there's no risk of other mutators changing it after selection. The targets in the configuration have no significance beyond this point.

E.g.
```go
// Target stores a snapshot of the target configuration when it was selected.
Target *config.Target
```

Reply:
Isn't it cleaner to just remove the side effect from `select_target`? Instead of just recording which target is selected it removes fields, which is a bit hard to discover and not really motivated in the code. Based on your comments, the motivation is to clean things up in order for consumption by summary/validate; shouldn't we just make that a separate step?

Reply:
What I see as a risk is that keeping them around means another location that new mutators can go and look at, even though everything under targets no longer has any effect. Variable interpolation won't run either, so values under it shouldn't be used.

I see how this is most convenient though. @andrewnester What do you think?

Reply:
It's just that over the past year I ran into this problem twice. You need to use a step through debugger to find where this property is secretly deleted. And the code that deletes it includes no rationale and is just meant to select the default target.

Reply:
Alright, I don't want to leave this PR open just because we can't make a decision here. I added 742210c8a6c9e99eee5b4f405dd0ff1fe666f8ff which merges the cleanup behavior back into SelectTarget and adds a few comments about the behavior for maintainers.